Origin: upstream
Applied-Upstream: Thu, 04 Nov 2010 20:36:50 +0000
Author: Flo Gravo <flo.gravo@gmail.com>
Description: Fix compilation with libnotify0.7

Upstream commit 6a964b0889fbdb164a036649c4aea1cc3c9818e3 Mon Sep 17 00:00:00 2001

Fixes bug #632848.
---
--- a/lib/engine/components/libnotify/libnotify-main.cpp
+++ b/lib/engine/components/libnotify/libnotify-main.cpp
@@ -158,7 +158,16 @@ LibNotify::on_notification_added (gmref_
 
   notif = notify_notification_new (notification->get_title ().c_str (),
 				   notification->get_body ().c_str (),
-				   urgency, NULL);
+				   urgency
+// NOTIFY_CHECK_VERSION appeared in 0.5.2 only
+#ifdef NOTIFY_CHECK_VERSION
+#if !NOTIFY_CHECK_VERSION(0,7,0)
+				     , NULL
+#endif
+#else
+				     , NULL
+#endif
+				   );
 
   g_signal_connect (notif, "closed",
 		    G_CALLBACK (on_notif_closed), notification.get ());
--- a/src/gui/main.cpp
+++ b/src/gui/main.cpp
@@ -2820,12 +2820,29 @@ ekiga_main_window_incoming_call_notify (
 
   body = g_strdup_printf ("%s\n%s\n%s", uri, app, account);
   
-  notify = notify_notification_new (title, body, GM_ICON_LOGO, NULL);
+
+  notify = notify_notification_new (title, body, GM_ICON_LOGO
+// NOTIFY_CHECK_VERSION appeared in 0.5.2 only
+#ifndef NOTIFY_CHECK_VERSION
+                                    , NULL
+#else
+#if !NOTIFY_CHECK_VERSION(0,7,0)
+                                    , NULL
+#endif
+#endif
+                                    );
   notify_notification_add_action (notify, "accept", _("Accept"), notify_action_cb, mw, NULL);
   notify_notification_add_action (notify, "reject", _("Reject"), notify_action_cb, mw, NULL);
   notify_notification_set_timeout (notify, NOTIFY_EXPIRES_NEVER);
   notify_notification_set_urgency (notify, NOTIFY_URGENCY_CRITICAL);
-  notify_notification_attach_to_status_icon (notify, statusicon);
+// NOTIFY_CHECK_VERSION appeared in 0.5.2 only
+#ifndef NOTIFY_CHECK_VERSION
+    notify_notification_attach_to_status_icon (notify, statusicon);
+#else
+#if !NOTIFY_CHECK_VERSION(0,7,0)
+    notify_notification_attach_to_status_icon (notify, statusicon);
+#endif
+#endif
   if (!notify_notification_show (notify, NULL)) {
     ekiga_main_window_incoming_call_dialog_show (mw, call);
   }
